includes/Core/Notifications/Notification.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Notifications\Notification
+ *
+ * @package   Google\Site_Kit\Core\Notifications
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Notifications;
+
+/**
+ * Class for representing a notification.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Notification {
+	/**
+	 * Unique notification slug.
+	 *
+	 * @since n.e.x.t
+	 * @var string
+	 */
+	private $slug;
+
+	/**
+	 * Notification arguments.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $args;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $slug Unique notification slug.
+	 * @param array  $args {
+	 *     Associative array of notification arguments.
+	 *
+	 *     @type string $title            Required notification title.
+	 *     @type string $content          Required notification content. May contain inline HTML tags.
+	 *     @type string $image            Image URL.
+	 *     @type string $cta_url          Call to action URL.
+	 *     @type string $cta_label        Call to action anchor text.
+	 *     @type string $cta_target       Call to action anchor target.
+	 *     @type string $learn_more_url   Learn more URL.
+	 *     @type string $learn_more_label Learn more anchor text.
+	 *     @type bool   $dismissible      Whether the notice should be dismissible. Default false.
+	 *     @type string $dismiss_label    Dismiss anchor text.
+	 * }
+	 */
+	public function __construct( $slug, array $args ) {
+		$this->slug = (string) $slug;
+		$this->args = array_merge(
+			array(
+				'title'            => '',
+				'content'          => '',
+				'image'            => '',
+				'cta_url'          => '',
+				'cta_label'        => '',
+				'cta_target'       => '',
+				'learn_more_url'   => '',
+				'learn_more_label' => '',
+				'dismissible'      => false,
+				'dismiss_label'    => __( 'Dismiss', 'google-site-kit' ),
+			),
+			$args
+		);
+	}
+
+	/**
+	 * Gets the notification's slug.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Unique notification slug.
+	 */
+	public function get_slug() {
+		return $this->slug;
+	}
+
+	/**
+	 * Prepares the JS representation of the Notification.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	public function prepare_for_js() {
+		return array(
+			'slug'           => $this->get_slug(),
```

Comment:
@aaemnnosttv That was not originally the plan, but maybe we should change this to `id` (just here, for JS)? "slug" is kind of the term only used in PHP, for consistency maybe `id` would be better for JS.

Reply:
Sure. I assumed it should be included since we need it for identifying the notice in the click handlers. I'll update the key to `id`.